packages/dds/map/src/directory.ts

```diff
@@ -693,17 +698,34 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         return this.localValueMaker.fromSerializable(serializable);
     }
 
+    private getParentDirectory(relativePath: string): IDirectory | undefined {
+        const absolutePath = this.makeAbsolute(relativePath);
+        const subdirs = absolutePath.substr(1).split(posix.sep);
```

Comment:
Do directory paths always start with /? Is this enforced somewhere?

Reply:
Note that string you get at the end does not have starting "/". So feels a bit dangerous to assume so, as other code may pass string without leading "/" (similar to how this code does), and I assume getWorkingDirectory() and other APIs are not enforcing that paths should start with slash

Reply:
BTW, your code could be much more efficient if you use lastIndexOf() and substr(), without ever splitting into chunks

Reply:
makeAbsolute handles that and make sure we start with "/". This is same logic as used in getWorkingDirectory. So when we call getWorkingDirectory, we don't need to make sure that it starts with "/".
Changed the code to use lastIndexOf.